### DISCUSSION TITLE
Support for PathLike objects

### DISCUSSION THREAD

**Discussion by VeckoTheGecko** [Parcels-code/Parcels#1221] [Ideas]
Running the following:
```py
from pathlib import Path
PLOT_FOLDER = Path("visualisations_particles")
...
pset.show(savefile=PLOT_FOLDER / 'particles', ...)
```

I get the following error.
```
     98     plt.savefig(savefile)
---> 99     logger.info('Plot saved to ' + savefile + '.png')
    100     plt.close()

TypeError: can only concatenate str (not "WindowsPath") to str
```

There are many ways of specifying paths in Python (strings, Path from pathlib and others). I feel that in general supporting these different ways of specifying paths would make parcels more flexible to users and their workflows.

This would be a more "quality of life" change, than a full blown feature. Regardless (as a fan of `pathlib`) I think this addition would be valuable, and would also bring `parcels` more in line with other libraries (eg. matplotlib) which support these other ways of specifying paths.

This also ties in with idea #1214, as having type annotations would make it clear the support for different parameter types (ie. that `savefile` only supports strings).

**Comment by erikvansebille** [comment]
Good suggestion, @VeckoTheGecko! I agree that we should support more pythonic workflows with path lib etc. 

Would changing the info-message to an f-string as done in PR #1222 fix this? Can you check?
